connector/src/main/java/org/jboss/as/connector/subsystems/jca/CachedConnectionManagerRemove.java

```diff
@@ -0,0 +1,28 @@
+package org.jboss.as.connector.subsystems.jca;
+
+import org.jboss.as.controller.AbstractRemoveStepHandler;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+
+/**
+ * @author <a href="ingo@redhat.com">Ingo Weiss</a>
+ */
+class CachedConnectionManagerRemove extends AbstractRemoveStepHandler {
+
+    static final CachedConnectionManagerRemove INSTANCE = new CachedConnectionManagerRemove();
+
+    private CachedConnectionManagerRemove() {}
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+
+        // We need to undefine any parameters and then set install to false, basically resetting the CCM settings
+        model.get(JcaCachedConnectionManagerDefinition.CcmParameters.DEBUG.getAttribute().getName()).set(ModelType.UNDEFINED);
```

Comment:
Just an FYI, this does not set the node to 'undefined', i.e. node.isDefined() would return 'true'. Rather, it sets the node value to the ModelType enum UNDEFINED.

A ModelNode can have a ModelType as its value. This is one of the things that allows us to describe the management API solely using ModelNode, e.g. we can describe the type of an attribute as ModelType.BOOLEAN etc.

If you want to make a node undefined, use either node.clear() or node.set(new ModelNode()).